src/ir.h

```diff
@@ -135,6 +141,9 @@ struct Expr {
   static Expr* CreateBrTable(VarVector* targets, Var default_target);
   static Expr* CreateCall(Var);
   static Expr* CreateCallIndirect(Var);
+  static Expr* CreateCatch(Var);
+  static Expr* CreateCatchAll();
+  static Expr* CreateCatchBlock(Expr* catch_, Expr* first);
```

Comment:
There shouldn't be a catch block at all. The problem was I was trying to find the right syntax (without parse conflicts) that I lost sight of how it should be structured. The simple (clean) solution appears to have the body of the "try" s-exp be a "block", and have the catches (including instruction list) appear after the block.